godotopenxrmeta/src/main/cpp/openxr_fb_render_model.cpp

```diff
@@ -0,0 +1,141 @@
+/* Copyright (c) 2022-present Godot XR contributors (see CONTRIBUTORS.md) */
+/*                                                                        */
+/* Original contributed implementation:                                   */
+/*   Copyright (c) 2022-2023 MattaKis Consulting Kft. (Migeran)           */
+/*                                                                        */
+/* Permission is hereby granted, free of charge, to any person obtaining  */
+/* a copy of this software and associated documentation files (the        */
+/* "Software"), to deal in the Software without restriction, including    */
+/* without limitation the rights to use, copy, modify, merge, publish,    */
+/* distribute, sublicense, and/or sell copies of the Software, and to     */
+/* permit persons to whom the Software is furnished to do so, subject to  */
+/* the following conditions:                                              */
+/*                                                                        */
+/* The above copyright notice and this permission notice shall be         */
+/* included in all copies or substantial portions of the Software.        */
+/*                                                                        */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,        */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF     */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. */
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY   */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,   */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE      */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                 */
+/**************************************************************************/
+
+#include "include/openxr_fb_render_model.h"
+
+#include "include/openxr_fb_render_model_extension_wrapper.h"
+
+#include <godot_cpp/classes/gltf_document.hpp>
+#include <godot_cpp/classes/gltf_state.hpp>
+#include <godot_cpp/classes/open_xr_interface.hpp>
+#include <godot_cpp/classes/xr_server.hpp>
+#include <godot_cpp/variant/utility_functions.hpp>
+
+using namespace godot;
+
+void OpenXRFbRenderModel::set_render_model_path(String p_render_model_path) {
+	render_model_path = p_render_model_path;
+	load_render_model();
+}
+
+String OpenXRFbRenderModel::get_render_model_path() {
+	return render_model_path;
+}
+
+TypedArray<String> OpenXRFbRenderModel::get_valid_paths() {
+	return OpenXRFbRenderModelExtensionWrapper::get_singleton()->get_valid_paths();
+}
+
+PackedStringArray OpenXRFbRenderModel::get_suggested_render_model_path_names() const {
+	PackedStringArray arr;
+	arr.push_back("/model_fb/controller/left");
+	arr.push_back("/model_fb/controller/right");
+	arr.push_back("/model_fb/keyboard/local");
+	arr.push_back("/model_fb/keyboard/remote");
```

Comment:
We should omit the keyboard paths until we add support for the keyboard extension. Maybe add a todo pointing to an issue to track.